holodeck/environments.py

```diff
@@ -305,3 +331,38 @@ def _prepare_agents(self, agent_definitions):
         if isinstance(agent_definitions, list):
             return [self._prepare_agents(x)[0] for x in agent_definitions]
         return [agent_definitions.type(client=self._client, name=agent_definitions.name)]
+
+    def _add_agents(self, agent_definitions):
+        """Add specified agents to the client. Set up their shared memory and sensor linkages.
+        Does not spawn an agent in the Holodeck, this is only for documenting and accessing already existing agents.
+        This is an internal function.
+
+        Positional Arguments:
+        agent_definitions -- The agent(s) to add.
+        """
+        if not isinstance(agent_definitions, list):
+            return self._add_agents([agent_definitions])
```

Comment:
Probably slightly faster to set agent_definitions = [agent_definitions]. Saves a function call and checking this again.